flip/uploader/session.go

```diff
@@ -15,42 +17,55 @@ type Session struct {
 
 	// Parts specifies the number of parts the file is supposed to be split into
 	// for uploading purposes.
-	Parts int `json:"parts"`
+	Parts int `json:"parts,string"`
 
 	// PartSize specifies the size of each part. The last part can be smaller
 	// than this value.
-	PartSize int64 `json:"part_size"`
+	PartSize int64 `json:"part_size,string"`
 
 	// MaxConnections specifies the maximum number of possible concurrent
 	// uploads.
-	MaxConnections int `json:"max_connections"`
+	MaxConnections int `json:"max_connections,string"`
+
+	ExtraFiles map[string]extra_files `json:"extra_files"`
```

Comment:
Lets say I am running a service that holds all my files on s3 bucket. I want to upload my video and extra files. The machine that is running this code has little storage. I can get an io.Reader for my S3 file and pass it to the Upload. Thanks to this I don't have to download anything to disk or to memory, because it's a data stream. I would expect the same behavior with extra files.